src/components/DataplanePolicies/DataplanePolicies.vue

```diff
@@ -172,9 +172,14 @@ export default {
           dppName: this.dppName,
         })
 
-        this.processItems(items)
-        this.hasItems = total > 0
-        this.items = items
+        // TODO: https://github.com/kumahq/kuma-gui/issues/351
```

Comment:
Can you just switch on the `kind` field here? That's the intention behind the tagged union.

Reply:
Theoretically, that could be done, but the current responses for other dataplanes don’t have a `kind` field (see https://github.com/kumahq/kuma-gui/issues/351#issuecomment-1216585108). I’m not sure that improves much.

Reply:
It definitely improves things, it's the semantics of the field that it's a sum type now and it's backwards compatible with `kind` being unset.

Consider what happens if another response type is added with an `items` array, then the code breaks again. The problem is not that `items` is or is not an array, it's that this code expects a specific variant which we can express by simply checking `.kind === "SidecarDataplane" || .kind === undefined`

Reply:
I'm pretty sure the current responses have a `kind` set and if not that's a bug in the API server.

Reply:
I will need to produce an actual response for different dataplane types to really understand this.

Currently, the mock data indicates a fundamentally different data structure. Checking for `king` being `undefined` would work the same way as checking for `items` being an array. In both cases, we can process `items`. However, I don’t know whether the presence of the `kind` field and it being a different value (e.g. `SidecarDataplane`) has a data structure that the current code can process. That’s why it seems very counter-intuitive to even look at the `kind` field. I extract it from the response and only do something with the `response` if it wasn’t there.

I’ll now try to get a different response from the actual API so I can be confident about the possible formats this endpoint can produce.

Reply:
We initially had one type of `Dataplane` and these `/policies` responses always had the same structure.

Then we introduced `MeshGateway` and now the response has either:

* `kind: SidecarDataplane` with the exact same structure as before
* `kind: MeshGatewayDataplane` with a new structure

The code (before the refactor) could process only the original structure and therefore it worked if and only if:

* `kind: undefined`/`kind` missing
* `kind: SidecarDataplane`

Reply:
> Checking for king being undefined would work the same way as checking for items being an array. In both cases, we can process items.

No, because the latter will lead to bugs again if further variants ever have an `items` array. Checking `kind` is more precise and explicit about what we're able to process

Reply:
Thanks for explaining this. It really wasn’t obvious to me that a response that _does_ have the `kind` field would _not_ come in the new format. I’ve updated the check as suggested.

Reply:
Yeah sorry, I realize I left that unsaid. Thanks

Reply:
Maybe there's something we can do with pulling the `pkg/api-server/testdata` in here as mock data to avoid the mock data getting out of date.